app/src/main/assets/web_extensions/webcompat_youtube/background.js

```diff
@@ -10,24 +11,44 @@ const targetUrls = [
  *    video pages intended for mobile phones, as linked from Google search results).
  */
 function redirectUrl(req) {
-	let redirect = false;
-	const url = new URL(req.url);
-	if (url.host.startsWith("m.")) {
-		url.host = url.host.replace("m.", "www.");
-		redirect = true;
-	}
-	if (!url.searchParams.get("disable_polymer")) {
-		url.searchParams.set("disable_polymer", "1");
-		redirect = true;
-	}
-	if (!redirect) {
-		return null;
-	}
-	return { redirectUrl: url.toString() };
+    let redirect = false;
+    const url = new URL(req.url);
+    if (url.host.startsWith("m.")) {
+        url.host = url.host.replace("m.", "www.");
+        redirect = true;
+    }
+    if (!url.searchParams.get("disable_polymer")) {
+        url.searchParams.set("disable_polymer", "1");
+        redirect = true;
+    }
+    if (!redirect) {
+        return null;
+    }
+    return { redirectUrl: url.toString() };
+}
+
+/**
+ * Override UA. This is required to get the equirectangular video formats from Youtube.
+ * Otherwise youtube uses custom 360 controls.
+ */
+function overrideUA(req) {
+    for (const header of req.requestHeaders) {
+        if (header.name.toLowerCase() === "user-agent") {
+            header.value = CUSTOM_USER_AGENT;
+        }
+    }
+    return { requestHeaders: req.requestHeaders };
```

Comment:
Is this to work around an issue with GV UA override?

Reply:
yes, it's the only way I found to apply the UA in youtube iframes

Reply:
Would you file a GV issue on bugzilla outlining the problems you encountered. It might end up being that webext is the preferred way to solve the problem but would like it to at least be on the teams radar.